tapcfg/server.go

```diff
@@ -231,16 +231,11 @@ func genServerConfig(cfg *Config, cfgLogger btclog.Logger,
 		}
 	}
 
-	// TODO(ffranr): This logic is leftover for integration tests which
-	//  do not yet enable a proof courier. Remove once all integration tests
-	//  support a proof courier.
-	var proofCourierCfg *proof.CourierCfg
-	if cfg.HashMailCourier != nil {
-		proofCourierCfg = &proof.CourierCfg{
-			ReceiverAckTimeout: cfg.HashMailCourier.ReceiverAckTimeout,
-			BackoffCfg:         cfg.HashMailCourier.BackoffCfg,
-			TransferLog:        assetStore,
-		}
+	proofCourierCfg := &proof.CourierCfg{
```

Comment:
This is the case of the itest failure; by default we run the tests with no proof courier:

https://github.com/lightninglabs/taproot-assets/blob/becddbf4a658ef3523ebbb4570daaaac1066b944/itest/tapd_harness.go#L244

So we panic here as the nil check was removed.

Reply:
Looks like itest now failing on a mistmatched state for an address receive, maybe the sideloading now causes the receive state to not be changed somewhere in the custodian?

```bash
assertions.go:769: 
        	Error Trace:	/home/jhb/tap/review_repo/tap/itest/assertions.go:769
        	            				/home/jhb/tap/review_repo/tap/itest/send_test.go:124
        	            				/home/jhb/tap/review_repo/tap/itest/test_harness.go:160
        	            				/home/jhb/tap/review_repo/tap/itest/integration_test.go:79
        	Error:      	Received unexpected error:
        	            	got status ADDR_EVENT_STATUS_TRANSACTION_CONFIRMED, wanted ADDR_EVENT_STATUS_COMPLETED
```

From `AssertNonInteractiveRecvComplete` in the itests calling the `AddrReceives` RPC.

Reply:
Related, we now have a name collision on these 'basic send unidirectional' itests which we should resolve.